### ISSUE TITLE
Train a logistic regression classifier

### ISSUE THREAD

**Issue by covuworie** [covuworie/nobel-physics-prizes#67]
Train a logistic regression classifier using the original features and the topics features with and without the sample weights. Select the following hyperparameters based on maximum MCC on the validation set:

- `penalty` which is used to specify whether the $L1$ or $L2$ norms are used in the regularization.
- `C`, the inverse of regularization strength.
- `class_weight`, the weights associated with the classes.
